score-xml/src/main/java/io/cloudslang/content/xml/services/ValidateService.java

```diff
@@ -2,57 +2,81 @@
+            } else {
+                xmlDocument = commonInputs.getXmlDocument();
+            }
 
+            if (Constants.XSD_PATH.equalsIgnoreCase(customInputs.getXsdDocumentSource())) {
+                Document doc = XmlUtils.createDocumentFromFile(customInputs.getXsdDocument(), commonInputs.getSecureProcessing());
+                xsdDocument = getStringValueOfDocument(doc);
+            } else {
+                xsdDocument = customInputs.getXsdDocument();
+            }
             XmlUtils.parseXML(xmlDocument, commonInputs.getSecureProcessing());
-            result.put(Constants.OutputNames.RETURN_RESULT, Constants.SuccessMessages.PARSING_SUCCESS);
+            result.put(Constants.Outputs.RETURN_RESULT, Constants.SuccessMessages.PARSING_SUCCESS);
 
-            if(xsdDocument != null && StringUtils.isNotBlank(xsdDocument)) {
-                validateAgainstXsd(xmlDocument, customInputs.getXsdDocument());
-                result.put(Constants.OutputNames.RETURN_RESULT, Constants.SuccessMessages.VALIDATION_SUCCESS);
+            if (xsdDocument != null && StringUtils.isNotBlank(xsdDocument)) {
+                validateAgainstXsd(xmlDocument, xsdDocument);
+                result.put(Constants.Outputs.RETURN_RESULT, Constants.SuccessMessages.VALIDATION_SUCCESS);
             }
 
-            result.put(Constants.OutputNames.RESULT_TEXT, Constants.SUCCESS);
+            result.put(Constants.Outputs.RESULT_TEXT, Constants.SUCCESS);
 
         } catch (SAXParseException e) {
-            result.put(Constants.OutputNames.RESULT_TEXT, Constants.FAILURE);
-            result.put(Constants.OutputNames.RETURN_RESULT, Constants.ErrorMessages.PARSING_ERROR + e.getMessage());
+            result.put(Constants.Outputs.RESULT_TEXT, Constants.FAILURE);
+            result.put(Constants.Outputs.RETURN_RESULT, Constants.ErrorMessages.PARSING_ERROR + e.getMessage());
         } catch (SAXException e) {
-            result.put(Constants.OutputNames.RESULT_TEXT, Constants.FAILURE);
-            result.put(Constants.OutputNames.RETURN_RESULT, Constants.ErrorMessages.VALIDATION_FAILURE + e.getMessage());
-        } catch (Exception e){
-            result.put(Constants.OutputNames.RESULT_TEXT, Constants.FAILURE);
-            result.put(Constants.OutputNames.RETURN_RESULT, e.getMessage());
+            result.put(Constants.Outputs.RESULT_TEXT, Constants.FAILURE);
+            result.put(Constants.Outputs.RETURN_RESULT, Constants.ErrorMessages.VALIDATION_FAILURE + e.getMessage());
+        } catch (Exception e) {
+            result.put(Constants.Outputs.RESULT_TEXT, Constants.FAILURE);
+            result.put(Constants.Outputs.RETURN_RESULT, e.getMessage());
         }
 
         return result;
     }
 
-    private static void validateAgainstXsd(String xmlDocument, String xsdDocument) throws Exception{
+    private static String getStringValueOfDocument(Document doc) throws TransformerException {
+        String xmlDocument;
```

Comment:
You could replace all these lines with only one:
return XmlUtils.getStringWriter(doc).toString();